### ISSUE TITLE
Loop example request

### ISSUE THREAD

**Issue by freewind** [scalamacros/xml-example#2]
Could you give an example that can loop on each element, and do some transformation?

Say, an XML:

``` xml
<div>
  <div class="header">This is header</div>
  <p class="body>This is body</p>
</div>
```

Is it possible to write a macro, which can add the `id` attribute to each tag? An also change the content of `This is header`.

 The final xml element will be:

```
<div id="1">
  <div  id="2" class="header">This is header !!!!</div>
  <p  id="3" class="body>This is body</p>
</div>
```

The id value is not important, I just want to find a way to do something to each element, use pattern matching, like:

```
def transform(elem:Tree): Tree = {
    elem match {
       case q"<div>...</div>" => ... 
       case q"<p>...</p>" => ...
    }
}
```

**Comment by freewind** [comment]
Just realised since we can get an instance of `Elem`, so is the way to iterate it (or transform it) the same as what we normally do? Like https://gist.github.com/zentrope/728034?

**Comment by densh** [comment]
Yes, it's the same. `Elem` is regular xml node and behaves like one. The only difference is that it also contains special `Unquote` parts which corresponds to Scala expressions within XML.
